### PR TITLE
Pinterest & Facebook Support

### PR DESCRIPTION
**Pull request by Smokietr**
@talyguryn

### CODE REVIEW

**Review by yosefanajjar — APPROVED** [review]

**Review by yosefanajjar — COMMENTED** [review]
This only handles facebook posts though, what about videos and comments?

**Comment by yosefanajjar** [comment]
Okay after testing this it works with videos and comments too, really impressed with your work! 👍 
image

The only catch as shown in the image is the spacing between the caption and the actual embedded content which should be a quick styles fix

File: src/services.js [hunk 1]
```diff
@@ -137,4 +137,20 @@ export default {
...
     id: ids => ids.join('/status/'),
   },
+  pinterest: {
+    regex: /https?:\/\/([^\/\?\&]*).pinterest.com\/pin\/([^\/\?\&]*)\/?$/,
+    embedUrl: 'https://assets.pinterest.com/ext/embed.html?id=<%= remote_id %>',
+    html: "<iframe scrolling='no' frameborder='no' allowtransparency='true' allowfullscreen='true' style='width: 100%; min-height: 400px; max-height: 1000px;'></iframe>",
+    id: function (e) {
+        return e[1];
+    }
+  },
```
Comment (yosefanajjar) [inline src/services.js:147]:
```suggestion
    id: (e) => {
        return e[1];
    }
  },
```

File: src/services.js [hunk 2]
```diff
@@ -137,4 +137,20 @@ export default {
...
+  facebook: {
+     regex: /https?:\/\/www.facebook.com\/([^\/\?\&]*)\/(.*)/,
+     embedUrl: 'https://www.facebook.com/plugins/post.php?href=https://www.facebook.com/<%= remote_id %>&width=500',
+     html: "<iframe scrolling='no' frameborder='no' allowtransparency='true' allowfullscreen='true' style='width: 100%; min-height: 500px; max-height: 1000px;'></iframe>",
+     id: function(e) {
+         return e.join('/');
+     }
```
Comment (yosefanajjar) [inline src/services.js:154]:
```suggestion
     id: (e) => {
         return e.join('/');
     }
```

**Comment by Smokietr** [comment]
Sorry. I have yet to find an answer to this question.

File: src/services.js [hunk 3]
```diff
@@ -137,4 +137,20 @@ export default {
...
     id: ids => ids.join('/status/'),
   },
+  pinterest: {
+    regex: /https?:\/\/([^\/\?\&]*).pinterest.com\/pin\/([^\/\?\&]*)\/?$/,
+    embedUrl: 'https://assets.pinterest.com/ext/embed.html?id=<%= remote_id %>',
+    html: "<iframe scrolling='no' frameborder='no' allowtransparency='true' allowfullscreen='true' style='width: 100%; min-height: 400px; max-height: 1000px;'></iframe>",
+    id: (e) => {
+        return e[1];
+    }
+  },
```
Comment (yosefanajjar) [inline src/services.js:147]: To be more consestant :)
```suggestion
    id: (ids) => {
        return ids[1];
    }
  },
```

File: src/services.js [hunk 4]
```diff
@@ -137,4 +137,20 @@ export default {
...
+  facebook: {
+     regex: /https?:\/\/www.facebook.com\/([^\/\?\&]*)\/(.*)/,
+     embedUrl: 'https://www.facebook.com/plugins/post.php?href=https://www.facebook.com/<%= remote_id %>&width=500',
+     html: "<iframe scrolling='no' frameborder='no' allowtransparency='true' allowfullscreen='true' style='width: 100%; min-height: 500px; max-height: 1000px;'></iframe>",
+     id: (e) => {
+         return e.join('/');
+     }
```
Comment (yosefanajjar) [inline src/services.js:154]:
```suggestion
     id: (ids) => {
         return ids.join('/');
     }
```

**Comment by yosefanajjar** [comment]
The space issues seem to be common take a look at this #37

**Comment by robonetphy** [comment]
Hi @Smokietr, Can you please add the corresponding test case inside [service.js](https://github.com/editor-js/embed/blob/master/test/services.js)

**Comment by Smokietr** [comment]
Hi @robonetphy Sorry for being late. I work intensely.

**Review by robonetphy — APPROVED** [review]
@Smokietr can you please update the version in the package.json from `2.3.1` -> `2.3.2`.
So we can merge the PR

File: package.json [hunk 5]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@editorjs/embed",
-  "version": "2.4.1",
+  "version": "2.3.2",
   "keywords": [
```
Comment (robonetphy) [inline package.json:4]: @Smokietr Please Change Accordingly.
Sorry for Earlier Typo which is `2.4.2`

**Review by robonetphy — CHANGES_REQUESTED** [review]

File: package.json (hunk 5 above)
Comment (robonetphy) [reply package.json:4]: And Also Please update [service.md](https://github.com/editor-js/embed/blob/master/docs/services.md)

**Comment by robonetphy** [comment]
@Smokietr, Also Please Resolve the conflicts and update to the next version accordingly and we ready to merge.

**Comment by robonetphy** [comment]
@Smokietr please give me access to your repo if you don't have time.Thank you

**Review by robonetphy — APPROVED** [review]

**Review by robonetphy — APPROVED** [review]

**Comment by robonetphy** [comment]
@Smokietr Thanks for your contribution.

**Pull request merged**